pyccel/codegen/printing/ccode.py

```diff
@@ -998,7 +1060,8 @@ def _print_Assign(self, expr):
         if isinstance(rhs, (NumpyFull)):
             code_init = ''
             if rhs.fill_value is not None:
-                code_init = 'array_fill({0}, {1});'.format(self._print(rhs.fill_value), lhs)
+                dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+                code_init = 'array_fill(({0}){1}, {2});'.format(dtype, self._print(rhs.fill_value), lhs)
```

Comment:
this lead to a double cast.
I think we better leave this without a fix until we find an appropriate one.

Reply:
Can you give code showing the double cast? Does this lead to a double cast only if `rhs.fill_value` is already a cast? Or always?

Reply:
Maybe the cast could be added only if `isinstance(rhs.fill_value, Literal)` ?

Reply:
I talked about that yesterday.
the test of `numpy.full` will not pass if I left it without adding cast.
because we have `literalInteger `printer don't add casting like `pythonInt` printer.